code/game/machinery/computer/abnormality_work.dm

```diff
@@ -140,6 +144,14 @@
 
 /obj/machinery/computer/abnormality/proc/finish_work(mob/living/carbon/human/user, work_type, pe = 0, max_pe = 0, work_speed = 2 SECONDS, training)
 	working = FALSE
+	if (pe >= datum_reference.success_boxes)
+		datum_reference.understanding = clamp((datum_reference.understanding + (datum_reference.max_understanding/10)), 0, datum_reference.max_understanding)
+	else
+		if (pe >= datum_reference.neutral_boxes)
+			datum_reference.understanding = clamp((datum_reference.understanding + (datum_reference.max_understanding/20)), 0, datum_reference.max_understanding)
+	if (!datum_reference.fully_understood && datum_reference.understanding == datum_reference.max_understanding)
+		datum_reference.fully_understood = TRUE
+		datum_reference.current.gift_chance *= 1.5
```

Comment:
I think this whole section should be moved to the datum reference.

Reply:
moved